clients/android/app/src/main/java/app/lockbook/loggedin/editor/HandwritingEditorActivity.kt

```diff
@@ -0,0 +1,218 @@
+package app.lockbook.loggedin.editor
+
+import android.os.Bundle
+import android.os.Handler
+import android.view.SurfaceHolder
+import android.view.View
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+import app.lockbook.R
+import app.lockbook.utils.*
+import com.beust.klaxon.Klaxon
+import kotlinx.android.synthetic.main.activity_handwriting_editor.*
+import java.util.*
+import timber.log.Timber
+
+class HandwritingEditorActivity : AppCompatActivity() {
+    private lateinit var handwritingEditorViewModel: HandwritingEditorViewModel
+    private lateinit var handwritingThread: Thread
+    private var timer: Timer = Timer()
+    private val handler = Handler()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_handwriting_editor)
+
+        val id = intent.getStringExtra("id")
+        val name = intent.getStringExtra("name")
+
+        if (id == null) {
+            errorHasOccurred("Unable to retrieve id.")
+            finish()
+            return
+        }
+
+        if (name == null) {
+            errorHasOccurred("Unable to retrieve name.")
+            finish()
+            return
+        }
+        handwriting_editor_drawing_name.text = name
+
+        handwritingEditorViewModel =
+            ViewModelProvider(
+                this,
+                HandwritingEditorViewModelFactory(application, id)
+            ).get(HandwritingEditorViewModel::class.java)
+
+        handwritingEditorViewModel.errorHasOccurred.observe(
+            this,
+            { errorHasOccurred ->
+                errorHasOccurred(errorHasOccurred)
+            }
+        )
+
+        if (startUpDrawing(id)) return
+
+        startBackgroundSave()
+        setUpHandwritingToolbar()
+    }
+
+    private fun startUpDrawing(id: String): Boolean {
+        val contents = handwritingEditorViewModel.handleReadDocument(id)
+
+        if (contents != null && contents.isNotEmpty()) {
+            val lockbookDrawable = if (handwritingEditorViewModel.lockBookDrawable == null) {
+
+                Klaxon().parse<Drawing>(contents)
+            } else {
+                handwritingEditorViewModel.lockBookDrawable
+            }
+
+            if (lockbookDrawable == null) {
+                errorHasOccurred("Unable to load this drawing.")
+                return true
+            }
+
+            handwriting_editor.lockBookDrawable = lockbookDrawable
+            handwriting_editor.holder.addCallback(object : SurfaceHolder.Callback {
+                override fun surfaceCreated(holder: SurfaceHolder?) {
+                    handwriting_editor.setUpBitmapDrawable()
```

Comment:
Perhaps this grouping logic could be better split between `Activity` and `View`, perhaps all the `init` steps that `View` requires could be grouped into 1 `View` function